### PR TITLE
Feat: add base resource section for component ref

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: Qiaozp <qiaozhongpei.qzp@alibaba-inc.com>### Description of your changes

Generated files is here: https://github.com/kubevela/kubevela.io/pull/928

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: references/docgen/markdown.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/oam-dev/kubevela/pkg/oam/discoverymapper"
```
Comment (Somefive) [inline references/docgen/markdown.go:28]: refactor the go import here?

**Review by Somefive — COMMENTED** [review]
Could you give some descriptions for the changed effect in this PR? In the PR description?

File: references/docgen/vars.go [hunk 2]
```diff
@@ -30,4 +30,6 @@ const (
 	Description = "Description"
 	// Examples marks the title of example in reference doc
 	Examples = "Examples"
+	// Base is the title of base in reference doc
+	Base = "Base on Kubernetes resource kind(s)"
```
Comment (wonderflow) [inline references/docgen/vars.go:34]: check my comment on : https://github.com/kubevela/kubevela.io/pull/928/files

**Review by wonderflow — APPROVED** [review]
great work

**Pull request merged**